bin/pull_condor_rpms.sh

```diff
@@ -75,6 +75,7 @@ for CONDOR_REPO in ${CONDOR_REPOS[@]}; do
   RSYNC_URL="$RSYNC_ROOT/$CONDOR_SERIES/$DVER/x86_64/$CONDOR_REPO/$SOURCE_SET*.rpm"
   echo "rsyncing $RSYNC_URL to $NEW_REPO_DIR"
   if ! rsync --times $RSYNC_URL $NEW_REPO_DIR --link-dest $CURRENT_REPO_DIR; then
-    echo "Warning: No packages found for $RSYNC_URL. Skipping"
+    echo "Error: No packages found for $RSYNC_URL."
```

Comment:
How do you know that the error is that no packages were found? rsync is a program that pulls things over the network and writes them to disk -- there are many other reasons that it could fail.  You don't have to enumerate them, since rsync should print the specific error, but you should change the error message to be more generic so it isn't misleading.

If you want to specifically check whether there are packages to pull first, then maybe do something with this:

> As a special case, if a single source arg is specified without a destination, the files are listed in an output format similar to "ls -l".

```bash
rsync_tmpfile=$(mktemp rsync.XXXXXX)
rsync "$RSYNC_URL" > "$rsync_tmpfile"
ret=$?
if [[ $ret != 0 ]]; then
    echo "Unable to get directory listing for $RSYNC_URL: rsync returned $ret"
    rm -f "$rsync_tmpfile"
    exit 1
elif [[ $(wc -l "$rsync_tmpfile") == 0 ]]; then
    echo "Directory listing for $RSYNC_URL returned no files"
    rm -f "$rsync_tmpfile"
    exit 1
fi
```
(untested)

Reply:
also you should capture and log the return code from the other `rsync` statement; it might be useful information.